src/webgl/p5.RendererGL.js

```diff
@@ -1913,23 +1923,49 @@ p5.RendererGL = class RendererGL extends p5.Renderer {
         defaultShaders.imageLightDiffusedFrag
       );
     }
+    // Create a shader for cubemap conversion
+    const cubemapShader = this._pInst.createShader(
```

Comment:
Just above this, we create and cache a shader so that we only ever need one copy. Can we add that caching here too?

Reply:
I have applied caching mechanism to possibly all the shaders. I am not sure whether this aligns with what you were suggesting through this @davepagurek .PLease have a look into the changes I've made.
Thanks.!

Reply:
I think we don't want to wrap all the existing shader sources in a caching function, since those are just the sources for one part of a shader (e.g. just the vertex or fragment source, and often not the full source either, since we need to add a prefix to it.) A caching function would be good for this `createShader` call (since you're still creating a new shader every time this line is hit, currently.) A few lines up, `this.diffusedShader` already does caching, but can be refactored to use the same caching system you use here if you're factoring that out into a common pattern.

Reply:
oh actually I see that you've added that in `_getCubemapShader`. I think rather than making a new shader here, we can just use `const cubemapShader = this._getCubemapShader()` instead.